src/hdf5_dl.c

```diff
@@ -97,44 +105,209 @@ hid_t H5E_ERR_CLS_g = -1;
 
 static bool is_init = false;
 
+/*
+ * Try to find a symbol within a library
+ * 
+ * handle: Handle to the library
+ * symbol: Symbol to look for
+ * Returns: a pointer to the symbol or NULL 
+ * if the symbol can't be found 
+ */
+void *find_sym(void *handle, const char *symbol) {
+
+  void *ret = NULL, *err = NULL;
+  dlerror(); /* clear error code */
+  ret = dlsym(handle, symbol);
+
+  if(ret != NULL && (err = dlerror()) == NULL)
+    return ret;
+  else
+    return NULL;
+}
+
+/*
+ * Check that all symbols have been loaded
+ * 
+ * Returns: 0 if an error occured, -1 for success
```

Comment:
This is the opposite of the standard C convention (where zero indicates success).  I'd consider reversing the logic, but if you'd prefer to keep it this way, you should at least point out that it's reversed.